src/components/VueGist.vue

```diff
@@ -43,11 +43,18 @@ export default {
       return `${this.url}?${this.params}`
     }
   },
-    created() {
-    this.getGistData(this.gistId);
+  watch: {
+    gistId: {
+      handler: function () {
+        this.$data.gistData = "loading...";
+        this.getGistData();
+      },
+      immediate:true,
+      deep: true,
+    },
```

Comment:
This looks good, but it got me thinking about the `file` prop. If someone dynamically changes gist id to another gist id which doesn't have the file name then it will break.

Do you think it might be useful to make `file` dynamic too? Or will it be wise to add additional checks on `file` to handle such cases?

Reply:
I disagree with mentioned above. Correct me if am wrong though... Filename is not a mandatory prop right? And the gist data is solely based on gistId right? As long as the user provides/changes the gistId it will always work like a charm 🥳

Reply:
Sorry for my communication. Look at the tests for better understanding.

Reply:
```
<template>
  <div 
    id="app" 
    class="div-style">
    <h3>Change the gist id dynamically</h3>
    <button 
      class="button" 
      @click="changeGistId">
      <span>Click Me</span>
    </button>
    <vue-embed-gist 
      :gist-id="gistIdValue" 
      file="index.js"
    />
  </div>
</template>

<script>
import VueEmbedGist from "../src";

export default {
  components: {
    VueEmbedGist,
  },
  data() {
    return {
      gistIdValue: "8bab656a910829ab9c32d7700c570be6",
    };
  },
  methods: {
    changeGistId() {
      console.log("changed gist id");
      if (this.$data.gistIdValue === "8bab656a910829ab9c32d7700c570be6") {
        this.$data.gistIdValue = "f361c725868e018f64901d483e4231b1";
      } else {
        this.$data.gistIdValue = "8bab656a910829ab9c32d7700c570be6";
      }
    },
  },
};
</script>

<style scoped>
.div-style {
  width: 50%;
  align-self: center;
  align-content: center;
  text-align: center;
  margin: 0 auto;
}
</style>
```
The scenario that I am talking about can be reproduced by this code. In this case on changing the gist id the dynamically, the gist won't load because the user provided a filename.

Reply:
I see 🤔, Is the Filename really necessary?.
file: {
      type: String,
      required: **false**,
      default: ""
    }.
We can load the gist data without giving the Filename, let me know your thoughts.

Reply:
I see 🤔 ill look into it., Do you have any idea about this use case, if so Let me know?
🙏 Sorry, I misunderstood the scenario @sudhanshu-15

Reply:
I am thinking multiple ways we could handle this:
- Restrict dynamically changing gist ids if component has a `file` prop.
- Allow changing file dynamically -> but this might introduce more issues.
- Load old gist if new gist id/file combination is invalid.
- Show an error display if at anytime we fail to load gist, will create a separate issue to address this.

For now I guess the path of least resistance would be restricting changing gist ids if there is a `file` prop. What do you think?

Reply:
I think option 4 is more reliable and proper solution and must needed also 🤔...

I'll create a separate PR for that one I have some idea for that let's see what we can do.

And I would like to mention 🤩 I really appreciate for immediate responses man thanks a lot.

Reply:
Oh for now I'll stop the method call if the element have file prop 👍🏻

Reply:
> I think option 4 is more reliable and proper solution and must needed also 🤔...
> 
> I'll create a separate PR for that one I have some idea for that let's see what we can do.
> 
> And I would like to mention 🤩 I really appreciate for immediate responses man thanks a lot.

😄 yes I have created an issue for point 4 along with some other improvements and also added those open for [Hacktoberfest](https://hacktoberfest.digitalocean.com/)

 I hope this has been a good learning for you as it has been for me.

Reply:
> Oh for now I'll stop the method call if the element have file prop 👍🏻

yes this works 👍🏽

Reply:
May I know what is Danger Expected — Waiting for status to be reported i don't understand

Reply:
Danger is https://danger.systems/js/

Automated reviews for trivia things. Very useful tool. Take a look :) If you have more questions let me know.